core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -279,13 +279,13 @@ class RequestSendThread(val controllerId: Int,
 
 }
 
-class ControllerBrokerRequestBatch(controller: KafkaController) extends  Logging {
+class ControllerBrokerRequestBatch(controller: KafkaController, eventManager: ControllerEventManager) extends  Logging {
```

Comment:
The intent is for `KafkaController.eventManager` to be visible for testing (there's a comment next to the field). Maybe we need to change the tests to access it reflectively since it's understandable that people may miss the comment.

The reasoning is that by passing the `KafkaController` everywhere, it makes it harder to understand and limit the responsibility of each class. It would be nicer if we passed more granular instances . For example, this class needs `eventManager`, `config`, `topicDeletionManager` and `controllerChannelManager`. If we passed those instead, we get a picture of what the class needs immediately instead of having to read every line of code.